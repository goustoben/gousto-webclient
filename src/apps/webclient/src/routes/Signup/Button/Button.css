@value Oyster, White, Bluecheese from 'styles/colors.css';

.cta {
  composes: fontStyleButtonL from 'design-language/typography.css';
  /**
   * min-height instead of height so when text breaks in 2 lines, then button would not lose it's paddings.
   */
  min-height: 3rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: Bluecheese;
  color: White;
  padding: 0.5rem;
  border-radius: 3px;
  cursor: pointer;

  &,
  &:hover {
    border: none;
  }
}

.isDisabled {
  background: Oyster;
  border-color: Oyster;
  color: White;
  cursor: not-allowed;

  &:hover {
    color: White;
  }
}

.isFullWidth {
  width: 100%;
}
