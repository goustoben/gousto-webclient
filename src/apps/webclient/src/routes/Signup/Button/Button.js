import React from 'react'

import actual from 'actual'
import classNames from 'classnames'
import PropTypes from 'prop-types'

import css from './Button.css'

const ctaText = (isLastStep) => {
  if (isLastStep) {
    return 'Confirm'
  }
  if (actual('width', 'px') <= 767) {
    return 'Next'
  }

  return 'Continue'
}

const Button = ({ children, onClick, isLastStep, ...buttonProps }) => {
  const buttonText = children || ctaText(isLastStep)

  const className = classNames(css.cta, {
    [css.isDisabled]: buttonProps.disabled,
    [css.isFullWidth]: buttonProps.width,
  })

  return (
    <button
      className={className}
      data-testing={buttonProps['data-testing']}
      disabled={buttonProps.disabled}
      onClick={onClick}
      onKeyDown={onClick}
      type="button"
    >
      {buttonText}
    </button>
  )
}

Button.propTypes = {
  isLastStep: PropTypes.bool.isRequired,
  onClick: PropTypes.func,
  children: PropTypes.node,
}

Button.defaultProps = {
  onClick: () => {},
  children: undefined,
}

export { Button }
