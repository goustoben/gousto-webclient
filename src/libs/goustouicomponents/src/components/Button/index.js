export { Button } from './Button.logic'
