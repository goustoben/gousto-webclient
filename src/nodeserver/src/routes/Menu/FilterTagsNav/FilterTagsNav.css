@value White from 'styles/colors.css';
@value ScreenSMMin from 'styles/screenSizes.css';

.filterTagContainer {
    composes: paddingMDTop paddingXLBottom from 'sass-loader!styles/spacing.scss';
    background: White;
    display: none;

    @media ScreenSMMin {
        display: flex;
    }
}

.filterListContainer {
    composes: pageContainer from 'styles/layout.css';
    width: 100%;
    display: flex;
    padding: 0 30px;
}

.filtersCTA {
    min-width: 12%;
}
