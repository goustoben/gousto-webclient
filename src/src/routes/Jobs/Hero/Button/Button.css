@value sizeLG, sizeXL from 'styles/typography.css';
@value Bluecheese, White from 'styles/colors.css';

.buttonContainer {
	composes: textCenter from 'styles/typography.css';
	composes: inlineBlock from 'styles/display.css';
	composes: fullWidth from 'styles/positioning.css';
}

.buttonInner {
	composes: inlineBlock from 'styles/display.css';
	margin: 0 auto;
}

.copy {
	composes: textBold textCenter from 'styles/typography.css';
	composes: paddingSMTop paddingZeroBottom paddingXXLLeftRight noMargin from 'sass-loader!styles/spacing.scss';
	color: Bluecheese;
	font-size: 24px;
	text-shadow: 0px 2px 5px White;
}

.chevron {
	composes: fa-basic-class  fa-chevron-down from 'styles/vendor/font-awesome-module.css';
	font-size: 24px;
	color: Bluecheese;
	composes: paddingXLBottom from 'sass-loader!styles/spacing.scss';
}
