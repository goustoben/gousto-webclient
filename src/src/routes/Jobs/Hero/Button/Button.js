import React from 'react'
import css from './Button.css'
import Link from 'Link'

const CTA = ({ children, link }) => (
	<div className={css.buttonContainer}>
		<Link to={link}>
			<div className={css.buttonInner}>
				<div className={css.buttonText}>
					<p className={css.copy} >{children}</p>
					<span className={css.chevron} aria-hidden="true" ></span>
				</div>
			</div>
		</Link>
	</div>
)

CTA.propTypes = {
  children: React.PropTypes.string,
  link: React.PropTypes.string,
}

export default CTA
