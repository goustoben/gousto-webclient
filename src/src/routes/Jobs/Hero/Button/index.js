export {CTA as Button} from './Button'
