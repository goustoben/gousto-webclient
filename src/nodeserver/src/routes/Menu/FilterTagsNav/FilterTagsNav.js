import React, { PropTypes } from 'react'
import Button from 'Button'
import FilterTagsList from './FilterTagsList'
import css from './FilterTagsNav.css'

const FilterTagsNav = ({ onCTAClick, tags }) => (
	<div className={css.filterTagContainer}>
		<div className={css.filterListContainer}>
			<div className={css.filtersCTA}>
				<Button fill width="full" onClick={onCTAClick}>
					Filters
				</Button>
			</div>
			<FilterTagsList tags={tags} />
		</div>
	</div>
)

FilterTagsNav.propTypes = {
	onCTAClick: PropTypes.func,
	tags: PropTypes.arrayOf(PropTypes.string),
}

FilterTagsNav.defaultProps = {
	tags: [],
	onCTAClick: () => {},
}

export default FilterTagsNav
