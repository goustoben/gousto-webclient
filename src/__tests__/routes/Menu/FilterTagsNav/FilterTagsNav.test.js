import React from 'react';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';
import Button from 'Button'
import FilterTagsNav from 'routes/Menu/FilterTagsNav/FilterTagsNav'

describe("FilterTag", () => {

  describe("rendering", () => {

    test("Will render FilterTag component", () => {
      const component = <FilterTagsNav />

      const tree = renderer
      .create(component)
      .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe("bahviour", () => {

    test("Will handle click on the CTA", () => {
      const clickHandler = jest.fn()
      const component = <FilterTagsNav onCTAClick={clickHandler} />
      const wrapper = shallow(component)

      wrapper.find(Button).first().simulate('click')

      expect(clickHandler).toHaveBeenCalled()
    })
  })
})
