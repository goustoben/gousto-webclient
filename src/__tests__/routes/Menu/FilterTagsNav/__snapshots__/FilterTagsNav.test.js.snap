// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTag rendering Will render FilterTag component 1`] = `null`;
